src/Client.php

```diff
@@ -19,9 +28,15 @@ public function __construct($client = null)
 		$this->client = $client;
 
 		if (is_null($client)) {
+			$stack = HandlerStack::create();
+			$stack->push(GuzzleRetryMiddleware::factory());
+
 			$this->client = new GuzzleClient([
+				'handler' => $stack,
 				'base_uri' => Smartpay::getApiUrl(),
 				'timeout'  => Smartpay::getPostTimeout(),
+				'max_retry_attempts' => 5,
+				'retry_on_status' => [500, 502, 503, 504]
```

Comment:
I think it is not a good way to handle 5xx errors in SDK. It might cause unnecessary traffic for the backend server which may cause more severe issues when the backend server is down.

Reply:
We want to provide a basic 3 to 5 retries on network error and some random error(basically it will be 5xx error, but when we do retry. The error gones.) Any other suggestion on this?

Reply:
Basically, we should not handle 5xx errors by design in SDK. If there are random errors, then fix them from backend logic. If we would like to provide this option for clients to retry some HTTP errors in SDK. We should expose it in the config and set the default to an empty array, and let the clients decide if they want to retry with the error codes that they set in the config.